evo_researcher/functions/web_search.py

```diff
@@ -1,9 +1,14 @@
 import os
+import tenacity
 from dotenv import load_dotenv
 from tavily import TavilyClient
 
 from evo_researcher.models.WebSearchResult import WebSearchResult
+from evo_researcher.functions.cache import persisted_inmemory_cache
 
+
```

Comment:
Are these benchmarking-specific decorators? I could see the argument for either way. If Nestor wants us to restrict them to something we only do for benchmarking, we could add a patch step, e.g. in benchmark.py, e.g.

```
from evo_researcher.functions import web_search

@tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_fixed(1), reraise=True)
@persisted_inmemory_cache
def patched_web_search(
    query: str, api_key: str, max_results=5
) -> list[web_search.WebSearchResult]:
    return web_search.web_search(query, api_key, max_results)


web_search.web_search = patched_web_search
```

Reply:
Good point, I guess they are? Both in-memory and file cache are local, so the agent can't add to them forever. 

So the agent needs to be executed with DISABLE_CACHE=1, or I can rather make DISABLE_CACHE=1 the default value, and we will need to enable it explicitly in our local environments.

I'd go with making DISABLE_CACHE=1 the default value, what do you think?

Reply:
Changed, check it out please 🙏

Reply:
`DISABLE_CACHE` -> `ENABLE_CACHE` 🔥🔥🔥 

Yeah happy with that default off behaviour. There is still the danger with benchmarking, now that we have 2 levels of cache granularity, that the user might:
- do `ENABLE_CACHE=1 python benchmark,py --cache-dir foo`
- now you have predictions, possibly with misleading timing numbers, saved to `foo`
- try to fix re-run to get correct timing numbers with `ENABLE_CACHE=0 python benchmark,py --cache-dir foo`
- but you've just loaded the same cached predictions from `foo`

Reply:
Then instead of adding warning to the benchmark, what about not saving the timing in the prediction at all, if caching is enabled?

That way, even if someone does `ENABLE_CACHE=0 python benchmark.py --cache-dir foo`, he will see timing missing and will know there is an issue.

Reply:
Ah yes, that makes sense. If `ENABLE_CACHE=1`, then the `Prediction` should have `p.time == None`. It's fine if the benchmark cache dir is set though.

Reply:
You could get cases then where some predictions have timing, and some do not, so we'll need to handle that

Reply:
Changed to `prediction.time = time.time() - start if not ENABLE_CACHE else None`

And the function is already doing 

```
def _compute_mean_time(
        self, predictions: t.List[Prediction], markets: t.List[Market]
    ):
        # Note: times are optional
        times = [p.time for p in predictions if p.time]
        if times:
            return sum(times) / len(times)
        else:
            return None
```

So it's okay if times will be None 👌